ArduCopter/mode_auto.cpp

```diff
@@ -343,63 +354,59 @@ void ModeAuto::land_start(const Vector2f& destination)
     copter.ap.prec_land_active = false;
 }
 
-// auto_circle_movetoedge_start - initialise waypoint controller to move to edge of a circle with it's center at the specified location
-//  we assume the caller has performed all required GPS_ok checks
-void ModeAuto::circle_movetoedge_start(const Location &circle_center, float radius_m)
+Mode::_LoiterTurns Mode::loiter_turns;
+
+void Mode::_LoiterTurns::initialise(const Location &circle_center_loc, float circle_radius_m, uint8_t turns)
 {
     // set circle center
-    copter.circle_nav->set_center(circle_center);
+    copter.circle_nav->set_center(circle_center_loc);
 
     // set circle radius
-    if (!is_zero(radius_m)) {
-        copter.circle_nav->set_radius(radius_m * 100.0f);
-    }
+    copter.circle_nav->set_radius_cm(circle_radius_m * 100.0f);
 
-    // check our distance from edge of circle
-    Vector3f circle_edge_neu;
-    copter.circle_nav->get_closest_point_on_circle(circle_edge_neu);
-    float dist_to_edge = (inertial_nav.get_position_neu_cm() - circle_edge_neu).length();
+    circle_center = circle_center_loc;
+    _total_turns = turns;
+    _state = _LoiterTurns::State::READY;
```

Comment:
Do you think we should just decide here-and-now whether we should move-to-edge or start circling - and eliminate the READY state?

Reply:
Done